json-rpc/src/methods.rs

```diff
@@ -80,6 +80,72 @@ async fn get_metadata(service: JsonRpcService, _: Vec<Value>) -> Result<BlockMet
     Ok(BlockMetadata { version, timestamp })
 }
 
+/// Returns transactions by range
+async fn get_transactions(
+    service: JsonRpcService,
+    params: Vec<Value>,
+) -> Result<Vec<TransactionView>> {
+    let p_start_version: Number = serde_json::from_value(params[0].clone())?;
+    let p_limit: Number = serde_json::from_value(params[1].clone())?;
+    let include_events: bool = serde_json::from_value(params[2].clone())?;
+
+    let start_version = p_start_version
+        .as_u64()
+        .ok_or(format_err!("Invalid start_version"))?;
+    let limit = p_limit.as_u64().ok_or(format_err!("Invalid limit"))?;
+
+    ensure!(
+        limit > 0 && limit <= 1000,
+        "limit must be smaller than 1000"
+    );
+
+    let txs = service.db.get_transactions(
+        start_version,
+        limit,
+        service.db.get_latest_version()?,
+        include_events,
+    )?;
+
+    let mut result = vec![];
+
+    for (v, tx) in txs.transactions.iter().enumerate() {
+        result.push(TransactionView {
+            version: start_version + v as u64,
+            transaction: tx.clone().into(),
```

Comment:
nit: do we need to really clone here? into_iter() doesn't work?

Reply:
fixed